### DISCUSSION TITLE
tailwind css is not working after deploy on vercel/netlify

### DISCUSSION THREAD

**Discussion by saiful-shaakil** [tailwindlabs/tailwindcss#8874] [Help] [answered]
Last 2 days, I'm trying to figure out the problem in my google-clone project. It is working fine on my localhost but not working fine after deploying it on vercel/netlify.
GitHub link of my project: https://github.com/saiful-shaakil/new-google-clone
Vercel live site link : https://google-clone-using-nextjs.vercel.app/

**Comment by ichenglin** [comment, marked as answer]
I'm not quite sure whether or not this was the issue,
but in **`tailwind.config.js`** you specified the **content path** to be **`"./components/**/*.{js,ts,jsx,tsx}"`**,
while the actual **source folder** was named **`Components`**.

Maybe try **[editing the fifth line of tailwind configuration file](https://github.com/saiful-shaakil/new-google-clone/blob/8107e69067a9fc3d0844bcbf2603cf242800fdfa/tailwind.config.js#L5)** and see if it works?
Navigate to **[tailwind.config.js](https://github.com/saiful-shaakil/new-google-clone/blob/main/tailwind.config.js)** and replace the line below
```js
"./components/**/*.{js,ts,jsx,tsx}",
```
with
```js
"./Components/**/*.{js,ts,jsx,tsx}",
```

**Reply by saiful-shaakil** [reply]
It is working now!
Thanks a lot man <3

**Reply by markjkaem** [reply]
Thanks

**Reply by AjayIrkal23** [reply]
thankyou brother

**Reply by alexastro01** [reply]
thank you so much

**Reply by Meriegg** [reply]
Thanks, didn't even realize that was my problem 💀

**Reply by nitesh98bhagat** [reply]
A BIG THANK YOU MATE ❤❤❤

**Reply by AdilAbbas135** [reply]
Thanks a lot sir

**Reply by Sufian-786-mumtaz** [reply]
Thanks sir

**Reply by venomkage** [reply]
Woah what a silly mistake I made! Thanks a lot, man! 🙇‍♂️

**Reply by sravankaruturi** [reply]
This was a silly mistake that I wouldn't have caught if not for this. Thank you so much!

**Reply by wogas** [reply]
Thank you so much.  How would I have thought of this when it works on my local windows machine. Perhaps, some Linux requirement

**Reply by Multichiki22** [reply]
Thank, you saved me

**Reply by underscore678** [reply]
Wow - This was my exact issue as well.

**Reply by F-47** [reply]
Life saver 🫡

**Reply by ahmod001** [reply]
i also did same mistake...thank u bro ...  you saved me

**Reply by tzhasan** [reply]
thank you sooo much

**Reply by Sushmzn9** [reply]
thanks life saviour

**Reply by sanny085** [reply]
Thanks it worked for me

**Reply by diptnc** [reply]
Thanks :)

**Reply by sanjeevkansyakar** [reply]
Thank you thank you thank you so much

**Reply by rehaanansari811** [reply]
thank you soo much brother you saved my internship.

**Reply by RommelGouala** [reply]
Nice. Thanks

**Reply by annedenicourt** [reply]
thank you so much !!! 
You saved my day ! The problem was just a cap... it is so a beginner's mistake :-)

**Reply by Karanjit08** [reply]
FOR ME NOT WORKING STILL

**Reply by rayenbenhassen07** [reply]
thank you so much bro u save my life !!!!1

**Reply by amanqureshi454** [reply]
thank you it works for me .

**Reply by SpooRe91** [reply]
Thanks, that solved the problem, adding a mental note :)

**Reply by pedrocnadais** [reply]
thanks bro, this also fixed my problem :)

**Reply by JAWAD-ASGHAR** [reply]
thanks for saving my days!

**Reply by HasiburRahman-hr11** [reply]
Thanks a lot ...

**Reply by ranjit800** [reply]
thanks a lot sir , i am also suffering 3 day

**Reply by SatyamDevv** [reply]
Thank you so much.

**Reply by ccz6ih** [reply]
With all the AI, All the Stack overflow this answer should be everywhere! It was my fix as well in 2025

**Comment by salimalsazu** [comment]
Thanks a lot. you made my day.  I was stuck for two days. I found this article and solved the problem.

**Comment by ankitbourasi0** [comment]
Thanks so  much its also work for me

**Comment by Shohanur-Shohan** [comment]
thank you so much  ..

**Comment by ngoyal100894** [comment]
lol. found the same issue. still sometimes you need google to resolve your issue. Otherwise, I was too dependent on chatGPT. Thanks a lot.

**Comment by wilkker** [comment]
Hey guys, good afternoon!, can anyone help me? I have the same problem, but I couldn't solve it :(


GitHub link of my project: https://github.com/wilkker/page_tailwindcss
Vercel live site link : https://pagina-npr5jc8v7-wilkker.vercel.app/

**Reply by wongjn** [reply]
Your live site link seems to be forcing me to login. Regardless, consider checking your paths. If your webroot is set to `src`, then the `<link>` to the CSS in `src/index.html` should be `href="/output.css"` or `href="./output.css"`:

```diff
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Page</title>
-    <link rel="stylesheet" href="/src/output.css">
+    <link rel="stylesheet" href="/output.css">

 </head>
```

**Comment by zainwaseem** [comment]
### nextjs 14
_if you are facing this issue in nextjs_

**Navigate to tailwind.config.ts and replace the line below and replace the content**
```
 content: [
    "./Components/**/*.{js,ts,jsx,tsx}",
    "./src/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
```

**Reply by DarkAsfu** [reply]
It's work

**Reply by ArshadSherif** [reply]
Its working now, thank you

**Reply by ahsantahirr** [reply]
Thank you brother

**Comment by PriyankaSoni03** [comment]
Hey guys, good afternoon!, can anyone help me? I am also having the same problem, but I couldn't solve it :(

GitHub link of my project: https://github.com/PriyankaSoni03/Razorpay-Clone
Vercel live site link : https://razorpay-clone-gz6j-git-main-priyankasoni03s-projects.vercel.app/

**Reply by diptnc** [reply]
@PriyankaSoni03  . the build command is missing from your package.json

**Reply by PriyankaSoni03** [reply]
Can you please specify that what command or lines should I add. As I am bit confused in this

**Reply by diptnc** [reply]
i added  this is package.json file
and then run `**npm run build**`

```
{
  "scripts": {
    "start": "vite",
    "build": "vite build"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.19",
    "postcss": "^8.4.38",
    "tailwindcss": "^3.4.1",
    "vite": "^5.2.6"
  }
}
```

After build process is completed , you will see dist folder .There you will see the build files that you can upload to any hosting. You can check tutorials on how to deploy static apps in vercel .

**Reply by PriyankaSoni03** [reply]
Thank You so much!! Its now working.

**Comment by SaqlainKazmi** [comment]
If you are using tailwind then check you folder name if it is camel case
then go to tailwind.config.ts and set your folder name this is find the
content and check the spelling (
content: [ './pages/**/*.{js,ts,jsx,tsx,mdx}',
'./Components/**/*.{js,ts,jsx,tsx,mdx}', './app/**/*.{js,ts,jsx,tsx,mdx}',
)

**Comment by mpal15** [comment]
hi all
i bulid the project react+taiwind but when i deploy the project tailwind is not working
here my tailwind config 
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./index.html",
    "./src/**/*.{js,jsx,ts,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
github link:https://github.com/mpal15/Book_explorer.git
vercel deploy link:https://book-explorer-woad.vercel.app/

**Reply by PriyankaSoni03** [reply]
Step 1: Add this in package.json file 
{
  "scripts": {
    "start": "vite",
    "build": "vite build"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.19",
    "postcss": "^8.4.38",
    "tailwindcss": "^3.4.1",
    "vite": "^5.2.6"
  }
}
Step 2: and then run **npm run build**
Step 3: After build process is completed , you will see "dist" named folder .There you will see the build files that you can upload to any hosting platform.

Alternative to the above steps is just host it on netlify, it will host your website directly by making dist file.
Hope this helps!!

**Comment by mpal15** [comment]
thank for help but is not working
github link:https://github.com/mpal15/Book_explorer.git
deploy link:https://book-explorer-red.vercel.app/

**Reply by diptnc** [reply]
check your tailwind installation , redo it , check after running npm run build  , then npm run preview if its working or not. Make sure you are importing all the necessary CSS files

**Comment by SaqlainKazmi** [comment]
The code is in your master branch vercel did’t  accept code code from
master brach push your code to main instead of master

**Comment by SaqlainKazmi** [comment]
If your css not working then (
If you are using tailwind then check you folder name if it is camel case
then go to tailwind.config.ts and set your folder name this is find the
content and check the spelling (
content: [ './pages/**/*.{js,ts,jsx,tsx,mdx}',
'./Components/**/*.{js,ts,jsx,tsx,mdx}', './app/**/*.{js,ts,jsx,tsx,mdx}',
)
)

**Comment by jbmarroq** [comment]
I had the same issue...named my folder Components and  had "./components/**/*.{js,ts,jsx,tsx}", in the config file, just changed it to a capital C and all worked upon deployment

**Comment by MagdaMie** [comment]
Hi all!
After deploying my page to netlify I ran into a few problems like 404 error, or a blank page being displayed - I managed them (one due to my repository being nested inside another folder, blank display - I removed the dist folder from git.ignore). The last one I can’t overcome - my tailwind/css isn’t applied at all.

Path to the css in index.html seems to be correct, the folder structure of dist too, all the tailwind and vite config as well as pacage.json seem to be correct too, all is up to date. Since my repository is nested inside another folder my build settings are as below:
<img width="811" alt="Screenshot 2024-12-25 at 20 05 39" src="https://github.com/user-attachments/assets/75247d72-59ea-454a-a765-ef2c15e257b2" />
It seems like I’ve tried everything and can not find the source of the issue, could you please take a look and help me?

Here is my repository: [GitHub - MagdaMie/website](https://github.com/MagdaMie/website/tree/master)
and here my website: https://profound-dolphin-3faf78.netlify.app/

**Comment by SudatiSimone** [comment]
Somebody can help me? Thanks
Same iussue :(

github link: https://github.com/SudatiSimone/sito-imgigi
deploy link: https://sito-imgigi.vercel.app/

**Comment by DarkAsfu** [comment]
If you need any support to solve the tailwind css issues kindly knock me. I
will try to solve your problem.


On Tue, Aug 5, 2025 at 10:44 PM Sudati Simone ***@***.***>
wrote:

> Somebody can help me? Thanks
> Same iussue :(
>
> github link: https://github.com/SudatiSimone/sito-imgigi
> deploy link: https://sito-imgigi.vercel.app/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tailwindlabs/tailwindcss/discussions/8874#discussioncomment-14009030>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5AKGVAVWASSLUSFEZEMKG33MDNO7AVCNFSM532DHJZ2U5DIOJSWCZC7NNSXTOSENFZWG5LTONUW63SDN5WW2ZLOOQ5TCNBQGA4TAMZQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 


Md. Ashraful Islam

MERN Stack Developer

Email: ***@***.***

Phone: +8801777112564

Dhaka, Bangladesh

<https://github.com/Darkasfu>  <https://www.linkedin.com/in/ashrafulislamm/>
  <https://mdashrafulislam.netlify.app/>

**Reply by rozsazoltan** [reply]
I'm building a UI package based on TailwindCSS that provides components for Astro/React/Svelte/Vue. How can I ensure that in projects using this package, the TailwindCSS build only scans the components from the Components folder that have actually been imported at least once in the project?

If I specify the node_modules/mypackage/components folder as a source, then both the Input and Button components end up in the final generated output - even if, for example, the Input component wasn't used a single time.

I'm interested in a solution compatible with both v3 and v4 for compatibility reasons.
